### PR TITLE
feat: add `useExportItemChat` hook

### LINKED ISSUE #202: Add Export chat hook
Expose new export chat endpoint defined in [PR#40](https://github.com/graasp/graasp-plugin-chatbox/pull/40)

### LINKED ISSUE #235: Move immutable types to `@graasp/sdk`

### PR DESCRIPTION
**Pull request by spaenleh**
This PR adds a hook to get the export chat data from the backend.

closes #202 
closes #235

### CODE REVIEW

**Review by pyphilia — CHANGES_REQUESTED** [review]
As mentioned, I think it will be wiser to export the chat as a file directly from the back with a mutation. The logic could be included in the export data functions (export action, export as zip) as well. To discuss with @juancarlosfarah and @dialexo .

**Comment by spaenleh** [comment]
I updated this PR with the deprecation of the immutable types exported by query-client.
These types were moved to `@graasp/sdk/frontend` in order to not duplicate the sources of typings.

closes #235

File: src/hooks/chat.test.ts [hunk 1]
```diff
@@ -81,19 +90,21 @@ describe('Chat Hooks', () => {
...
   describe('useItemChat with arguments', () => {
     const itemId = ITEMS.first()!.id;
+    const mockMessage: ChatMessage = {
+      id: 'some-messageId',
+      chatId: itemId,
+      body: 'some content',
+      creator: 'some-user',
+      createdAt: 'some Date',
+      updatedAt: 'some other Date',
+    };
```
Comment (pyphilia) [inline src/hooks/chat.test.ts:100]: Have a builder function to create a mock message?

**Review by pyphilia — APPROVED** [review]
🍾

**Pull request merged**